### PR TITLE
Run eslint/tslint on staged files and ignore removed files

### PR DESCRIPTION
**Pull request by blink1073**
## References
Partial fix for #7939.

## Code changes
Improved lint-staged file.

## User-facing changes
Better experience for contributors, who don't have to run the linter manually after the fact.

## Backwards-incompatible changes
None

### CODE REVIEW

File: lint-staged.config.js [hunk 1]
```diff
@@ -1,14 +1,34 @@
...
     return [
       `prettier --write ${escapedFileNames}`,
       `git add ${escapedFileNames}`
     ];
+  },
+  '**/*{.ts,.tsx}': filenames => {
+    const escapedFileNames = filenames
+      .filter(filename => fs.existsSync(filename))
+      .map(filename => `"${isWin ? filename : escape([filename])}"`)
+      .join(' ');
```
Comment (ajbozarth) [inline lint-staged.config.js:20]: Given this same "line" is used in all three cases it's probably better to make it a const function and call it in each instead.

File: lint-staged.config.js [hunk 2]
```diff
@@ -1,14 +1,34 @@
...
+    return [`tslint --fix ${escapedFileNames}`, `git add ${escapedFileNames}`];
```
Comment (ajbozarth) [inline lint-staged.config.js:21]: Based on what the `lint` call in package.json does I think updating this to: 
```suggestion
    return [
      `prettier --write ${escapedFileNames}`,
      `tslint --fix ${escapedFileNames}`,
      `git add ${escapedFileNames}`
    ];
```
and then removing `.ts,.tsx` from the next call would be more consistent.

**Comment by blink1073** [comment]
Thanks for the feedback @ajbozarth!  Updated.

**Comment by blink1073** [comment]
The `lint` scripts don't specify which files to lint, we use `.eslintignore` to screen.  Cheers!

**Pull request merged**